forge-gui/res/cardsfolder/upcoming/the_caves_of_androzani.txt

```diff
@@ -0,0 +1,11 @@
+Name:The Caves of Androzani
+ManaCost:3 W
+Types:Enchantment Saga
+K:Saga:4:DBStunCounter,DBAdditionalCounter,DBAdditionalCounter,DBSearch
+SVar:DBStunCounter:DB$ PutCounter | ValidTgts$ Creature.tapped | TgtPrompt$ Select un two two target tapped creature | CounterType$ STUN | CounterNum$ 2 | TargetMin$ 0 | TargetMax$ 2 |  SpellDescription$ Put two stun counters on each of up to two target tapped creatures.
+SVar:DBAdditionalCounter:DB$ PutCounter | CounterType$ ExistingCounter | Choices$ Permanent.nonSaga | ChoiceAmount$ Count$Valid Permanent.nonSaga | EachExistingCounter$ True | SpellDescription$ For each non-Saga permanent, choose a counter on it. You may put an additional counter of that kind on that permanent.
```

Comment:
this looks kind of wrong

Reply:
Is there an actual mechanism here for choosing the counter that gets added? Also, I'm not seeing a mechanism here for placing the counter to be optional.

This seems like a bit of a nightmare actually, because as-written, you're forced to choose a counter for every permanent that has a counter on it, then choose FOR EVERY SINGLE ONE OF THOSE PERMANENTS whether to add another. At least if a permanent has only one kind of counter on it, that decision gets eliminated. But if you have a bunch of tokens with +1/+1 counters on them, you have to make the second choice for each one of those. That's a LOT of choice-making.